pyfa_tool/dataset.py

```diff
@@ -0,0 +1,797 @@
+
+
+            # Find the 2d-whitelist fields
+            all_available_fields = FA._list_all_fieldnames_as_2d_fields()
+            subset_fields['2d_white'] = [field for field in whitelist if field in all_available_fields]
+
+            # Find the 3d-whitelist fields
+            all_available_3dfields = FA._list_all_3d_fieldnames_as_basenames()
+            subset_fields['3d_white'] = [field for field in whitelist if field in all_available_3dfields]
+
+            # Check at leas one field is included in the whitelists
+            if ((len(subset_fields['2d_white']) == 0) & (len(subset_fields['3d_white']) == 0)):
+                sys.exit(f'None of these fields are found in the FA file: {whitelist}')
+        else:
+            subset_fields['2d_white'] = FA._list_all_2d_fieldnames_as_2d_fields()
+            subset_fields['3d_white'] = FA._list_all_3d_fieldnames_as_basenames()
+
+        # ---------- Blacklist creation -------------------
+
+        if not (blacklist is None):
+            # test if the blacklist is an iterable (list/series/array):
+            if isinstance(blacklist, str):
+                blacklist = [blacklist]
+            if not isinstance(blacklist, Iterable):
+                sys.exit(f'{blacklist} is not an Iterable (like a list/array/...)')
+
+            # Get all available fields (if not yet made in the whitelist part)
+            if FA is None:
+                FA = FaFile(self.fafile)
+
+
+            # Find the 2d-blacklist fields
+            all_available_fields = FA._list_all_fieldnames_as_2d_fields()
+            subset_fields['2d_black'] = [field for field in blacklist if field in all_available_fields]
+
+            # Find the 3d-blacklist fields
+            all_available_3dfields = FA._list_all_3d_fieldnames_as_basenames()
+            subset_fields['3d_black'] = [field for field in blacklist if field in all_available_3dfields]
+
+            # Check at leas one field is included in the blacklists
+            if ((len(subset_fields['2d_white']) == 0) & (len(subset_fields['3d_white']) == 0)):
+                print(f'WARNING: None of these fields are found in the FA file: {blacklist}')
+
+
+        # create at tmpdir if not provided
+        tmpdir = IO.create_tmpdir(location=os.getcwd())
+
+
+        # There is no clean way i found to parse multiple lists as arguments
+        #for an R script. So write them to json, and read them in the Rscript.
+
+        Rfa_attr_json=os.path.join(tmpdir, 'Rfa_extra_attrs.json')
+        IO.write_json(datadict=subset_fields,
+                      jsonpath=Rfa_attr_json,
+                      force=True)
+
+        # Run Rscript to generete json files with data and meta info
+        r_script = os.path.join(package_path, 'modules',
+                                'rfa_scripts', 'get_all_fields.R')
+
+
+        convert_fa = subprocess.Popen([os.path.join(IO._get_rbin(), 'Rscript'), r_script,
```

Comment:
Error for 3D Fields that do not have all 46 levels.
```bash
data.import_fa(whitelist=["RAYT THER CL"])
```
An Error Occurred for this 3d field
<simpleError in dimnames(result) <- list(x = NULL, y = NULL, hybrid = minlev:maxlev): length of 'dimnames' [3] not equal to array extent>
```error
An Error Occurred for this 3d field
<simpleError in dimnames(result) <- list(x = NULL, y = NULL, hybrid = minlev:maxlev): length of 'dimnames' [3] not equal to array extent>
```
A correct warning is provided that fields S002-S045RAYT THER CL not found. But the resulting array has no data variables.

Reply:
Converted to issue #41